morgoth/addons/api/serializers.py

```diff
@@ -21,21 +31,46 @@ class Meta:
 
 
 class AddonGroupSerializer(serializers.ModelSerializer):
-    addons = serializers.PrimaryKeyRelatedField(many=True, queryset=Addon.objects.all(),
-                                                required=False)
+    addons = AddonSerializer(many=True, read_only=True)
+    addon_ids = serializers.ListField(required=False, write_only=True)
+    built_in_addons = AddonSerializer(many=True, read_only=True)
+    qa_addons = AddonSerializer(many=True, read_only=True)
+    shipped_addons = AddonSerializer(many=True, read_only=True)
+    browser_version = VersionField()
 
     class Meta:
         model = AddonGroup
         fields = (
             'id',
             'browser_version',
             'addons',
+            'addon_ids',
             'built_in_addons',
             'qa_addons',
             'shipped_addons',
         )
-        read_only_fields = (
-            'built_in_addons',
-            'qa_addons',
-            'shipped_addons',
-        )
+
+    def get_addons(self, obj):
+        return AddonSerializer(obj.addons, many=True).data
+
+    def create(self, validated_data):
+        addon_ids = validated_data.pop('addon_ids', [])
+
+        group = AddonGroup.objects.create(**validated_data)
+
+        for addon_id in addon_ids:
```

Comment:
Is this guaranteed to run atomically? If not, you should wrap this so that any failure results in the multiple saves happening here being rolled back.